CA_DataUploaderLib/MCUBoard.cs

```diff
@@ -273,16 +273,27 @@ private async Task<bool> ReadSerialNumber()
             }
 
             port.WriteLine("Serial");
-            var stop = DateTime.Now.AddSeconds(5);
             bool sentSerialCommandTwice = false;
             bool ableToRead = false;
             var finishedReadingHeader = false;
-            while (!finishedReadingHeader && DateTime.Now < stop)
+
+            // A cancellation token is made here rather than a simple timer since the ReadAsync function can hang
+            // if a device never sends a line for it to read.
+            int millisecondsTimeout = 3000;
```

Comment:
to avoid extra testing needed I suggest to stick to 5 seconds here, so that we are not changing the max time allowed to read the header during initialization (we had some old boards that were slow to initialize and respond with a header).